components/announcement_bar/cloud_trial_ended_announcement_bar/index.tsx

```diff
@@ -0,0 +1,118 @@
+import {makeGetCategory} from 'mattermost-redux/selectors/entities/preferences';
+import {savePreferences} from 'mattermost-redux/actions/preferences';
+import {
+    getCurrentUser,
+} from 'mattermost-redux/selectors/entities/users';
+
+const CloudTrialEndAnnouncementBar: React.FC = () => {
+    const usageDeltas = useGetUsageDeltas();
+    const limits = useGetLimits();
+    const subscription = useGetSubscription();
+    const getCategory = makeGetCategory();
+    const dispatch = useDispatch();
+    const preferences = useSelector((state: GlobalState) =>
+        getCategory(state, Preferences.FREEMIUM_TRIAL_END_BANNER),
+    );
+    const currentUser = useSelector((state: GlobalState) =>
+        getCurrentUser(state),
+    );
+
+    const shouldShowBanner = () => {
+        if (!subscription) {
+            return false;
+        }
+
+        // Make sure limits are loaded before showing banner
+        if (!limits || !limits[1]) {
+            return false;
+        }
+
+        if (!preferences) {
+            return false;
+        }
+        if (preferences.some((pref) => pref.name === CloudBanners.HIDE && pref.value === 'true')) {
+            return false;
+        }
+
+        const isFreeTrial = subscription.is_free_trial === 'true';
+        if (isFreeTrial) {
+            return false;
+        }
+
+        const trialEnd = new Date(subscription.trial_end_at * 1000);
+        const now = new Date();
+
+        // trial_end_at values will be 0 for all freemium subscriptions after June 15
+        // Subscriptions created prior to that will almost always have a trial_end_at value, guaranteed.
+        if (subscription.trial_end_at === 0 || trialEnd > now || trialEnd < new Date('2022-06-15')) {
+            return false;
+        }
+        if (!isSystemAdmin(currentUser.roles)) {
+            return false;
+        }
+        return true;
+    };
+
+    if (!shouldShowBanner()) {
+        return null;
+    }
+
+    const handleClose = () => {
+        dispatch(
+            savePreferences(currentUser.id, [
+                {
+                    category: Preferences.FREEMIUM_TRIAL_END_BANNER,
+                    user_id: currentUser.id,
+                    name: CloudBanners.HIDE,
+                    value: 'true',
+                },
+            ]),
+        );
+    };
+
+    let message = {
+        id: t('freemium.banner.trial_ended.premium_features'),
+        defaultMessage:
+            'Your trial has ended. Upgrade to regain access to paid features',
+    };
+    if (anyUsageDeltaValueIsNegative(usageDeltas)) {
+        message = {id: t('freemium.banner.trial_ended.archived_data'), defaultMessage: 'Your trial has ended. Upgrade to regain access to archived data'};
+    }
+
+    return (
+        <AnnouncementBar
+            type={AnnouncementBarTypes.CRITICAL}
+            showCloseButton={true}
+            onButtonClick={() => {
+                // TODO: Open the new Pricing modal.
```

Comment:
I think this can be updated now to the new pricing modal introduced in: https://github.com/mattermost/mattermost-webapp/pull/10271

Reply:
added